Add limit props to set height and width of XLine and YLine respectively

tanayv opened the pull request:
Adds the height/width limit props discussed in #149 

<img width="1142" alt="Screen Shot 2019-06-10 at 3 41 10 AM" src="https://user-images.githubusercontent.com/14301342/59181856-e2977a00-8b85-11e9-8f60-ed51af71c780.png">
<img width="1142" alt="Screen Shot 2019-06-10 at 3 40 43 AM" src="https://user-images.githubusercontent.com/14301342/59181858-e2977a00-8b85-11e9-8a43-53c660fb3ed0.png">

tanayv commented:
Thanks for the comments. I'll add the tests and address the comments with an updated PR soon!

tanayv commented on src/XLine.js:
```diff
@@ -47,13 +59,21 @@ export default class XLine extends React.Component {
+    let y2 = height + spacingBottom;
+
+    if (yLimit) {
+      y1 = yScale(yDomain[0]);
```
<img width="1135" alt="Screen Shot 2019-06-12 at 11 23 21 PM" src="https://user-images.githubusercontent.com/14301342/59375040-e3d5cc00-8d6a-11e9-87c0-2d577c0300cb.png">

So with y1 set as -spacingTop, the line begins from the top instead of the bottom. I assumed since the value is within the domain, so the line should begin from the lower bound of the domain. Let me know if that's not what you had in mind/if there's a better implementation for this.

krissalvador27 replied:
Gotcha! I think we're missing the spacingBottom as well so we need to account for that if users pass that in. Is it possible, for consistency, for the YLine to always grow from the top or bottom (the latter makes more sense to me). So if there's a yLimit the only thing that would be altered is y2 where y1 should stay `yScale(yDomain[0]) - spacingBottom`?

tanayv replied:
Hey @krissalvador27, sorry for the delay in making these changes. Finally had the chance to get to them. 

For the XLine to always consistently grow from the bottom (ie. by fixing the value of y1 as `yScale(yDomain[0]) - spacingBottom`), I will also have to account for the XLines created by the XGrid, which is a child of the XAxis component. 

As of now, the XAxis component doesn't accept the yDomain as a prop from the XYPlot, and so the XLines that make the grid do not have the prop yDomain available. So should I go ahead and add yDomain and yScale as a prop for both XAxis and XGrid to make all lines (including the grid) lines grow from the bottom?

krissalvador27 replied:
Hey @tanayv!

Thanks for catching that! XYPlot should hand yDomain and yScale down to all its children, so XAxis should easily accept it. I do worry that the complexity and scope is growing though. I think we can go with your original logic to accomplish the original XLine and YLine height and width limits and consistently have XLines grow from the top/limit point. 👍

tanayv commented:
Hey @krissalvador27, 

I've added the test cases for the new props, and fixed the undefined check on XLine. For the logic, I stuck to having only the lines with xLimit in their props to grow from the bottom, while all other XLines  consistently grow from the top. Also, I set `y1 = yScale(yDomain[0]) + spacingBottom` since now y1 is the lower point of the line. 

Thanks!
